### PR TITLE
Fix #3083: CertificateException due to PEM being decoded in CertUtils

### LINKED ISSUE #3083: CertificateException due to PEM being decoded in CertUtils
It looks like `CertUtils#getInputStreamFromDataOrFile` determines whether the certificate data is base64 encoded or not by checking if the data can be base64 decoded. However since okio considers the minus sign to be a valid character
([reference](https://github.com/square/okio/blob/465c07e7e98b091d39600a6720d8bf3c1121da34/okio/src/commonMain/kotlin/okio/-Base64.kt#L65)), even PEM data can be base64 decoded.

Observed the issue at [OpenIDConnectionUtils#getOIDCProviderTokenEndpointAndRefreshToken](https://github.com/fabric8io/kubernetes-client/blob/e8109a80f786a4d2a68b08be2f9efc94ffff5019/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/utils/OpenIDConnectionUtils.java#L356), which calls [getSSLContext](https://github.com/fabric8io/kubernetes-client/blob/e8109a80f786a4d2a68b08be2f9efc94ffff5019/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/utils/OpenIDConnectionUtils.java#L281), where the certificate data is [decoded to PEM](https://github.com/fabric8io/kubernetes-client/blob/e8109a80f786a4d2a68b08be2f9efc94ffff5019/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/utils/OpenIDConnectionUtils.java#L286) already; the PEM is then passed to [SSLUtils#trustManagers](https://github.com/fabric8io/kubernetes-client/blob/e8109a80f786a4d2a68b08be2f9efc94ffff5019/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/internal/SSLUtils.java#L97), then [CertUtils#createTrustStore](https://github.com/fabric8io/kubernetes-client/blob/e8109a80f786a4d2a68b08be2f9efc94ffff5019/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/internal/CertUtils.java#L80), and eventually [CertUtils#getInputStreamFromDataOrFile](https://github.com/fabric8io/kubernetes-client/blob/e8109a80f786a4d2a68b08be2f9efc94ffff5019/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/internal/CertUtils.java#L61). `ByteString.decodeBase64(data)` returns non-null value, therefore the PEM is decoded again.

Also observed the same issue when passing PEM data to `ConfigBuilder#withCaCertData`

## Reproduce
Use one of the DigiCert root CAs as an example
```
-----BEGIN CERTIFICATE-----
MIIDxTCCAq2gAwIBAgIQAqxcJmoLQJuPC3nyrkYldzANBgkqhkiG9w0BAQUFADBs
MQswCQYDVQQGEwJVUzEVMBMGA1UEChMMRGlnaUNlcnQgSW5jMRkwFwYDVQQLExB3
d3cuZGlnaWNlcnQuY29tMSswKQYDVQQDEyJEaWdpQ2VydCBIaWdoIEFzc3VyYW5j
ZSBFViBSb290IENBMB4XDTA2MTExMDAwMDAwMFoXDTMxMTExMDAwMDAwMFowbDEL
MAkGA1UEBhMCVVMxFTATBgNVBAoTDERpZ2lDZXJ0IEluYzEZMBcGA1UECxMQd3d3
LmRpZ2ljZXJ0LmNvbTErMCkGA1UEAxMiRGlnaUNlcnQgSGlnaCBBc3N1cmFuY2Ug
RVYgUm9vdCBDQTCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEBAMbM5XPm
+9S75S0tMqbf5YE/yc0lSbZxKsPVlDRnogocsF9ppkCxxLeyj9CYpKlBWTrT3JTW
PNt0OKRKzE0lgvdKpVMSOO7zSW1xkX5jtqumX8OkhPhPYlG++MXs2ziS4wblCJEM
xChBVfvLWokVfnHoNb9Ncgk9vjo4UFt3MRuNs8ckRZqnrG0AFFoEt7oT61EKmEFB
Ik5lYYeBQVCmeVyJ3hlKV9Uu5l0cUyx+mM0aBhakaHPQNAQTXKFx01p8VdteZOE3
hzBWBOURtCmAEvF5OYiiAhF8J2a3iLd48soKqDirCmTCv2ZdlYTBoSUeh10aUAsg
EsxBu24LUTi4S8sCAwEAAaNjMGEwDgYDVR0PAQH/BAQDAgGGMA8GA1UdEwEB/wQF
MAMBAf8wHQYDVR0OBBYEFLE+w2kD+L9HAdSYJhoIAu9jZCvDMB8GA1UdIwQYMBaA
FLE+w2kD+L9HAdSYJhoIAu9jZCvDMA0GCSqGSIb3DQEBBQUAA4IBAQAcGgaX3Nec
nzyIZgYIVyHbIUf4KmeqvxgydkAQV8GK83rZEWWONfqe/EW1ntlMMUu4kehDLI6z
eM7b41N5cdblIZQB2lWHmiRk9opmzN6cN82oNLFpmyPInngiK3BD41VHMWEZ71jF
hS9OMPagMRYjyOfiZRYzy78aG6A9+MpeizGLYAiJLQwGXFK3xPkKmNEVX58Svnw2
Yzi9RKR/5CYrCsSXaQ3pjOLAEFe4yHYSkVXySGnYvCoCWw9E1CAx2/S6cCZdkGCe
vEsXCS+0yx5DaMkHJ8HSXPfqIbloEpw8nL+e/IBcm2PN7EeqJSdnoDfzAIJ9VNep
+OkuE6N36B9K
-----END CERTIFICATE-----
```
Pass it to `ByteString#decodeBase64`, get non-null result.

### PR DESCRIPTION
**Pull request by rohanKanojia**
## Description
Fix #3083
Use `java.util.Base64` instead of `okio.ByteString` to decode base64 encoded
string since former one throws IllegalArgumentException in case input
string in invalid base64 encoded string.

`okio.ByteString` returns non-null value even for invalid base64 decoded
strings

Signed-off-by: Rohan Kumar <rohaan@redhat.com>

## Type of change

 - [X] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [X] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [X] I Added [CHANGELOG](../CHANGELOG.md) entry regarding this change
 - [X] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/internal/CertUtils.java [hunk 1]
```diff
@@ -284,10 +277,21 @@ public static KeyStore createKeyStore(String clientCertData, String clientCertFi
...
       if (line.indexOf(endMarker) != -1) {
-        return ByteString.decodeBase64(buf.toString()).toByteArray();
+        return Base64.getDecoder().decode(buf.toString());
       }
       buf.append(line.trim());
     }
     throw new IOException("PEM is invalid : No end marker");
   }
+
+  private static ByteArrayInputStream createInputStreamFromBase64DecodedString(String data) {
```
Comment (manusa) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/internal/CertUtils.java:287]: nit: shouldn't this be named `createInputStreamFromBase64EncodedString` or `createDecodedInputStreamFromBase64EncodedString` ?

File: kubernetes-client/src/test/resources/ssl/valid-non-decoded-cert.pem [hunk 2]
```diff
@@ -0,0 +1,23 @@
+-----BEGIN CERTIFICATE-----
```
Comment (manusa) [inline kubernetes-client/src/test/resources/ssl/valid-non-decoded-cert.pem:1]:
```
Common Name: DigiCert High Assurance EV Root CA
Issued By: DigiCert Inc
Issuing Certificate: DigiCert High Assurance EV Root CA
Serial Number: 02AC5C266A0B409B8F0B79F2AE462577
Signature: sha1WithRSAEncryption
Valid From: 00:00:00 10 Nov 2006
Valid To: 00:00:00 10 Nov 2031
Key Usage: Digital Signature, Certificate Sign, CRL Sign
Basic Constraints: CA:TRUE
Subject Key Identifier: B1:3E:C3:69:03:F8:BF:47:01:D4:98:26:1A:08:02:EF:63:64:2B:C3
Authority Key Identifier: keyid:B1:3E:C3:69:03:F8:BF:47:01:D4:98:26:1A:08:02:EF:63:64:2B:C3
```

Comment (manusa) [reply kubernetes-client/src/test/resources/ssl/valid-non-decoded-cert.pem:1]: I guess this file should be valid-non-base64-encoded-cert.pem

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/internal/CertUtils.java (hunk 1 above)
Comment (manusa) [reply kubernetes-client/src/main/java/io/fabric8/kubernetes/client/internal/CertUtils.java:287]: Maybe just `createInputStream` and specify in the javadoc that this method returns a decoded inputstream for the provided optionally base64 encoded String

File: kubernetes-client/src/test/resources/ssl/valid-non-base64-encoded-cert.pem [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+iBJMW5Y1jGBpQCmaPeoXIe6MlCvsguTOL048EjqU73BHoO0KiAvrs6Bo/p+CBzqE
+XlpwnupHYrNQot27nPCAeVxL0Qi7qRa+dFXFLVDeOgOl6kCINcc48nzeDbTA57Ya
+eniqG9vYP3EYL49OEQVv2MalTps9D6UiLqh56e28FRjLBaFLrOBbCGufoXkFnE+T
+PrVt5ldiwBwqFYdodv3Zv6NpK7xTHpPcCjLx3xSMFvgF0Jmp8QnvCUS633uGtagB
+aHwcDXdT0XbXqQLwK5pftM5nDfSCMjjcr/A293orfbHPWu4cc0UaCLsNZBFKdMc6
+xeBxp2xTi0j4c/+d/xIr2uO/TZPs9CkqUhGzLhLGS3k2XE/YpNxT74bhGADhDpjq
+9KzJAcudOUjKAjpeLP236gkf8mdanabQOnI8NNbFN/Zi+lQafsE9pEwr5lTePMQM
+UWvw7UeqCGZviiYzo3NHUJOy6HSxkNhNJ0H6bx81znmfIMNNnvQs0GKcFN7TTS7C
+iljS6PbiKUIwTo2q/rS3hKZT9cmt1+K7t9freBoNhO5kPCmHL6eQkqXJy6abvQ==
+-----END CERTIFICATE-----
```
Comment (manusa) [inline kubernetes-client/src/test/resources/ssl/valid-non-base64-encoded-cert.pem:28]:
```
Common Name: Non-Unique
Issuing Certificate: Non-Unique
Serial Number: F8A6D461BAD84ED0
Signature: sha256WithRSAEncryption
Valid From: 22:02:38 03 Jul 2019
Valid To: 22:02:38 01 Jul 2024
```

**Review by manusa — APPROVED** [review]
LGTM

**Pull request merged**